google/cloud/options_test.cc

```diff
@@ -328,6 +328,22 @@ TEST(OptionsSpan, Basics) {
   EXPECT_FALSE(internal::CurrentOptions().has<IntOption>());
 }
 
+TEST(FetchOption, Basics) {
+  auto opts = Options{}.set<StringOption>("foo").set<IntOption>(42);
+  EXPECT_FALSE(opts.has<BoolOption>());
+  EXPECT_TRUE(opts.has<IntOption>());
+  EXPECT_TRUE(opts.has<StringOption>());
+
+  auto b = internal::FetchOption<BoolOption>(opts);
+  EXPECT_FALSE(b.has_value());
```

Comment:
Optional optional nit:  For what its worth, the GoogleTest Matchers Reference says, "for testing whether an `optional<>` is set, check for equality with `nullopt`".  I don't know why.